core/update.php

```diff
@@ -283,61 +285,120 @@ function update_results_page() {
-  $output .= "<li><strong>Config files:</strong> Back up the entire directory at '" . $config_dir . "'.</li>\n";
-  $output .= "</ul>\n";
-  $output .= '<li>Put your site into <a href="' . base_path() . '?q=admin/config/development/maintenance">maintenance mode</a>.</li>' . "\n";
-  $output .= "<li>Install your new files into the appropriate location, as described in <a href=\"https://backdropcms.org/upgrade\">the handbook</a>.</li>\n";
-  $output .= "</ol>\n";
-  $output .= "<p>After performing the above steps proceed using the continue button.</p>\n";
+  if (!$updates) {
+    backdrop_set_message('No pending updates.');
+
+    $output .= theme('links', array('links' => update_helpful_links()));
+
+    // No updates to run, so caches won't get flushed later. Clear them now.
+    backdrop_flush_all_caches();
+  }
+  else {
+    backdrop_set_message('Back up your database and site before you continue. <a href="https://docs.backdropcms.org/backup">Learn how</a>.', 'warning');
+
+    $elements = backdrop_get_form('update_script_overview_form');
+    $output .= backdrop_render($elements);
+  }
+
+  // If this is an upgrade from Drupal 7, check if additional modules need to be
+  // enabled, and present them in a list.
   $module_status_report = update_upgrade_check_dependencies();
 	if (!empty($module_status_report)) {
     $output .= $module_status_report;
   }
-  $form_action = check_url(backdrop_current_script_url(array('op' => 'selection', 'token' => $token)));
-  $output .= '<form method="post" action="' . $form_action . '">
-  <div class="form-actions">
-    <input type="submit" value="Continue" class="form-submit button-primary" />
-    <a href="' . base_path() . '">Cancel</a>
-  </div>
-  </form>';
-  $output .= "\n";
+
   return $output;
 }
 
+/**
+ * Form constructor for the "Overview" step of update.php.
+ *
+ * This form is an intermediate step in the database/configuration update script
+ * workflow, that is shown after the "Verify requirements" step (see
+ * update_check_requirements()). The purpose of this page is to encourage taking
+ * a backup of the database and the site, and to give the opportunity to put the
+ * site in maintenance mode. After this step, the "Review updates" step follows
+ * (see update_selection_page()).
+ *
+ * @see update_script_overview_form_submit()
+ * @ingroup forms
+ */
+function update_script_overview_form($form, &$form_state) {
+  $maintenance_mode = state_get('maintenance_mode');
+  // Store the current maintenance mode status so we can restore it when done.
+  $_SESSION['maintenance_mode'] = $maintenance_mode;
+
+  if ($maintenance_mode) {
+    backdrop_set_message(t('The site is already in maintenance mode.'));
+  }
+
+  $form['maintenance_mode'] = array(
+    '#type' => 'checkbox',
+    '#id' => 'maintenance_mode',
+    '#title' => 'Perform updates with site in maintenance mode (<strong>strongly recommended</strong>)',
+    '#default_value' => TRUE,
+    '#disabled' => $maintenance_mode,
```

Comment:
Why would users be allowed to decide if putting the site in maintenance mode when it is not already in maintenance mode, but they cannot decide about that when the site is already in maintenance mode?  
The checkbox should be always enabled.